config/bootstrap.php

```diff
@@ -29,6 +29,19 @@ class_alias('Cake\Mailer\Email', 'Cake\Network\Email\Email');
 class_alias('Cake\Mailer\Transport\MailTransport', 'Cake\Network\Email\MailTransport');
 class_alias('Cake\Mailer\Transport\SmtpTransport', 'Cake\Network\Email\SmtpTransport');
 
+// @deprecated Backwards compatibility with earler 3.x versions.
+class_alias('Cake\Http\Client', 'Cake\Network\Http\Client');
+class_alias('Cake\Http\Client\CookieCollection', 'Cake\Network\Http\CookieCollection');
+class_alias('Cake\Http\Client\FormData', 'Cake\Network\Http\FormData');
+class_alias('Cake\Http\Client\Message', 'Cake\Network\Http\Message');
+class_alias('Cake\Http\Client\Request', 'Cake\Network\Http\Request');
+class_alias('Cake\Http\Client\Response', 'Cake\Network\Http\Response');
+class_alias('Cake\Http\Client\Adapter\Stream', 'Cake\Network\Http\Adapter\Stream');
+class_alias('Cake\Http\Client\Auth\Basic', 'Cake\Network\Http\Auth\Basic');
+class_alias('Cake\Http\Client\Auth\Digest', 'Cake\Network\Http\Auth\Digest');
+class_alias('Cake\Http\Client\Auth\Oauth', 'Cake\Network\Http\Auth\Oauth');
+class_alias('Cake\Http\Client\FormDataPart', 'Cake\Network\Http\FormData\Part');
```

Comment:
Do you mind adding `false` as a third argument to all `class_alias` function calls in this file? I think That will prevent the classes from being loaded without a reason, although I'm not 100% sure that is the effect it will have.

I'm just concerned that we are loading a large part of the framework on every request